pkg/apis/serving/v1alpha2/kfservice_types.go

```diff
@@ -182,17 +183,21 @@ type CustomSpec struct {
 	Container v1.Container `json:"container"`
 }
 
+// EndpointStatusMap defines the observed state of KFService endpoints
+type EndpointStatusMap map[constants.KFServiceEndpoint]*StatusConfigurationSpec
```

Comment:
We probably want to specialize the StatusConfiguration spec to the endpoint

i.e. 
`type EndpointStatusMap map[constants.KFServiceEndpoint]*EndpointStatus`

The StatusConfigurationSpec might be used elsewhere?

Thinking a little deeper, does it make sense to do:
```
type StatusConfigurationSpec struct {
  map[constants.KFServiceEndpoint]*EndpointStatusSpec
  .. potentially other status configuration fields?
}
```
Finally, I want to protest the name "endpoint" for all of these things. When I think endpoint, I think "addressable unit", but that isn't necessarily the case. I feel fairly strongly that "component" is a better word for this.

Reply:
Do you mean rename StatusConfigurationSpec to EndpointStatus?... 

For Endpoint vs Component.. They are different addressable endpoints, its just that you may not want to address them directly? In fact the transformer does indeed reference the predictor endpoint?

Reply:
My poor jetlagged brain was very wrong. I spent some time looking though and you're 100% right (i was thinking predictor/explainer were endpoints, not default/canary)